ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Update.hs

```diff
@@ -172,23 +167,16 @@ pureTryAddTx ::
      )
   => LedgerCfg (LedgerState blk)
      -- ^ The ledger configuration.
-  -> (GenTx blk -> TxSizeInBytes)
-     -- ^ The function to claculate the size of a transaction.
   -> WhetherToIntervene
   -> GenTx blk
      -- ^ The transaction to add to the mempool.
   -> InternalState blk
      -- ^ The current internal state of the mempool.
   -> TryAddTx blk
-pureTryAddTx cfg txSize wti tx is
-    -- We add the transaction if there is at least one byte free left in the
-    -- mempool, and...
-  | let curSize = msNumBytes  $ isMempoolSize is
-  , curSize < getMempoolCapacityBytes (isCapacity is)
-    -- ... if the mempool has less than 2.5 mebibytes of ref scripts.
-  , let maxTotalRefScriptSize = 5 * 512 * 1024 -- 2.5 Mebibytes
-        curTotalRefScriptSize = isTotalRefScriptSize is
-  , curTotalRefScriptSize Prelude.< maxTotalRefScriptSize
+pureTryAddTx cfg wti tx is
+    -- We add the transaction if it wouldn't overrun any component of the
+    -- mempool capacity.
+  | TxSeq.toSize (isTxs is) `Measure.plus` sz Measure.<= isCapacity is
```

Comment:
The previous logic (add the transaction if there is at least one byte free left in the mempool) was added for fairness reasons: #21 (larger issue #50) (also: https://github.com/IntersectMBO/ouroboros-network/pull/4436#discussion_r1159512582)

I don't know how crucial this change was for overall mempool fairness; there was a specific scenario that prompted #50, but I don't remember the details.

Reply:
Do you agree with my reasoning here? https://github.com/IntersectMBO/ouroboros-consensus/issues/21#issuecomment-2229479007

Reply:
Yes, I can't think of any particular benefit of #21 given the use of the `MVar` :+1: Would be nice to get that confirmed eventually by Duncan/Damian (if only to understand the context back then), but this doesn't have to block this PR.